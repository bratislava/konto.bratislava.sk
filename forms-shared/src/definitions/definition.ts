import { Schemas } from '../generator/functions'
import stanoviskoKInvesticnemuZameru from './stanovisko-k-investicnemu-zameru'
import predzahradky from './predzahradky'
import priznanieKDaniZNehnutelnosti from './priznanie-k-dani-z-nehnutelnosti'
import { generalTermsAndConditions, taxTermsAndConditions } from './terms-and-conditions'
import zavazneStanoviskoKInvesticnejCinnosti from './zavazne-stanovisko-k-investicnej-cinnosti'
import komunitneZahrady from './komunitne-zahrady'

enum FormDefinitionType {
  SlovenskoSk = 'SlovenskoSk',
  Tax = 'Tax',
  Email = 'Email',
}

type FormDefinitionBase = {
  slug: string
  schemas: Schemas
  title: string
  termsAndConditions: string
}

type FormDefinitionSlovenskoSk = FormDefinitionBase & {
  type: FormDefinitionType.SlovenskoSk | FormDefinitionType.Tax
  pospID: string
  pospVersion: string
}

type FormDefinitionEmail = FormDefinitionBase & {
  type: FormDefinitionType.Email
  email: string
}

type FormDefinition = FormDefinitionSlovenskoSk | FormDefinitionEmail

export const formDefinitions: FormDefinition[] = [
  {
    type: FormDefinitionType.SlovenskoSk,
    title: 'Žiadosť o stanovisko k investičnému zámeru',
    slug: 'stanovisko-k-investicnemu-zameru',
    schemas: stanoviskoKInvesticnemuZameru,
    pospID: '00603481.zavazneStanoviskoKInvesticnejCinnosti',
    pospVersion: '0.7',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    title: 'Žiadosť o záväzné stanovisko k investičnej činnosti',
    slug: 'zavazne-stanovisko-k-investicnej-cinnosti',
    schemas: zavazneStanoviskoKInvesticnejCinnosti,
    pospID: '00603481.stanoviskoKInvesticnemuZameru',
    pospVersion: '0.7',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    title: 'Predzáhradky',
    slug: 'predzahradky',
    schemas: predzahradky,
    pospID: '00603481.predzahradky',
    pospVersion: '1.0',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    title: 'Komunitné záhrady',
    slug: 'predzahradky',
    schemas: komunitneZahrady,
    pospID: '00603481.komunitneZahrady',
    pospVersion: '1.0',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.Tax,
    title: 'Priznanie k dani z nehnuteľností',
    slug: 'priznanie-k-dani-z-nehnutelnosti',
    schemas: priznanieKDaniZNehnutelnosti,
    pospID: 'esmao.eforms.bratislava.obec_024',
    pospVersion: '201501.2',
    termsAndConditions: taxTermsAndConditions,
  },
]
