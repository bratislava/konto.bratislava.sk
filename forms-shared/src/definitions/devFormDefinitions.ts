import { FormDefinition } from './formDefinitionTypes'

export const devFormDefinitions: FormDefinition[] = []
