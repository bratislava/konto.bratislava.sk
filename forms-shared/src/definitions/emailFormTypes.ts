export enum MailgunTemplateEnum {
  NASES_SENT = 'NASES_SENT',
  GINIS_DELIVERED = 'GINIS_DELIVERED',
  NASES_GINIS_IN_PROGRESS = 'NASES_GINIS_IN_PROGRESS',
  GINIS_SUCCESS = 'GINIS_SUCCESS',
  GINIS_DENIED = 'GINIS_DENIED',
  ATTACHMENT_VIRUS = 'ATTACHMENT_VIRUS',
  OLO_NEW_SUBMISSION = 'OLO_NEW_SUBMISSION',
  OLO_SENT_SUCCESS = 'OLO_SENT_SUCCESS',
  TSB_NEW_SUBMISSION = 'TSB_NEW_SUBMISSION',
  TSB_SENT_SUCCESS = 'TSB_SENT_SUCCESS',
}
