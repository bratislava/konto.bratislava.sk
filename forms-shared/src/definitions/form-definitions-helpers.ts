import {
  FormDefinition,
  FormDefinitionEmail,
  formDefinitions,
  FormDefinitionSlovenskoSk,
  FormDefinitionSlovenskoSkGeneric,
  FormDefinitionSlovenskoSkTax,
  FormDefinitionType,
} from './form-definitions'

export const isSlovenskoSkGenericFormDefinition = (
  formDefinition: FormDefinition,
): formDefinition is FormDefinitionSlovenskoSkGeneric =>
  formDefinition.type === FormDefinitionType.SlovenskoSkGeneric

export const isSlovenskoSkTaxFormDefinition = (
  formDefinition: FormDefinition,
): formDefinition is FormDefinitionSlovenskoSkTax =>
  formDefinition.type === FormDefinitionType.SlovenskoSkTax

export const isSlovenskoSkFormDefinition = (
  formDefinition: FormDefinition,
): formDefinition is FormDefinitionSlovenskoSk =>
  isSlovenskoSkGenericFormDefinition(formDefinition) ||
  isSlovenskoSkTaxFormDefinition(formDefinition)

export const isEmailFormDefinition = (
  formDefinition: FormDefinition,
): formDefinition is FormDefinitionEmail => formDefinition.type === FormDefinitionType.Email

export const getFormDefinitionBySlug = (slug: string): FormDefinition | null => {
  const formDefinition = formDefinitions.find((formDefinition) => formDefinition.slug === slug)
  return formDefinition ?? null
}
