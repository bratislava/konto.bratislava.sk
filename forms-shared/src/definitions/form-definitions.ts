import { Schemas } from '../generator/functions'
import stanoviskoKInvesticnemuZameru from './stanovisko-k-investicnemu-zameru'
import predzahradky from './predzahradky'
import priznanieKDaniZNehnutelnosti from './priznanie-k-dani-z-nehnutelnosti'
import { generalTermsAndConditions, taxTermsAndConditions } from './terms-and-conditions'
import zavazneStanoviskoKInvesticnejCinnosti from './zavazne-stanovisko-k-investicnej-cinnosti'
import komunitneZahrady from './komunitne-zahrady'

enum FormDefinitionType {
  SlovenskoSk = 'SlovenskoSk',
  Tax = 'Tax',
  Email = 'Email',
}

type FormDefinitionBase = {
  slug: string
  title: string
  schemas: Schemas
  termsAndConditions: string
}

type FormDefinitionSlovenskoSk = FormDefinitionBase & {
  type: FormDefinitionType.SlovenskoSk | FormDefinitionType.Tax
  pospID: string
  pospVersion: string
}

type FormDefinitionEmail = FormDefinitionBase & {
  type: FormDefinitionType.Email
  email: string
}

type FormDefinition = FormDefinitionSlovenskoSk | FormDefinitionEmail

export const formDefinitions: FormDefinition[] = [
  {
    type: FormDefinitionType.SlovenskoSk,
    slug: 'stanovisko-k-investicnemu-zameru',
    title: 'Žiadosť o stanovisko k investičnému zámeru',
    schemas: stanoviskoKInvesticnemuZameru,
    pospID: '00603481.zavazneStanoviskoKInvesticnejCinnosti',
    pospVersion: '0.7',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    slug: 'zavazne-stanovisko-k-investicnej-cinnosti',
    title: 'Žiadosť o záväzné stanovisko k investičnej činnosti',
    schemas: zavazneStanoviskoKInvesticnejCinnosti,
    pospID: '00603481.stanoviskoKInvesticnemuZameru',
    pospVersion: '0.7',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    slug: 'predzahradky',
    title: 'Predzáhradky',
    schemas: predzahradky,
    pospID: '00603481.predzahradky',
    pospVersion: '1.0',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.SlovenskoSk,
    slug: 'komunitne-zahrady',
    title: 'Komunitné záhrady',
    schemas: komunitneZahrady,
    pospID: '00603481.komunitneZahrady',
    pospVersion: '1.0',
    termsAndConditions: generalTermsAndConditions,
  },
  {
    type: FormDefinitionType.Tax,
    title: 'Priznanie k dani z nehnuteľností',
    slug: 'priznanie-k-dani-z-nehnutelnosti',
    schemas: priznanieKDaniZNehnutelnosti,
    pospID: 'esmao.eforms.bratislava.obec_024',
    pospVersion: '201501.2',
    termsAndConditions: taxTermsAndConditions,
  },
]
