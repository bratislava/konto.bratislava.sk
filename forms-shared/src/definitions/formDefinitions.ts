import stanoviskoKInvesticnemuZameru from '../schemas/stanoviskoKInvesticnemuZameru'
import predzahradky from '../schemas/predzahradky'
import priznanieKDaniZNehnutelnosti from '../schemas/priznanieKDaniZNehnutelnosti'
import { generalTermsAndConditions, taxTermsAndConditions } from './termsAndConditions'
import zavazneStanoviskoKInvesticnejCinnosti from '../schemas/zavazneStanoviskoKInvesticnejCinnosti'
import komunitneZahrady from '../schemas/komunitneZahrady'
import { FormDefinition, FormDefinitionType } from './formDefinitionTypes'

export const formDefinitions: FormDefinition[] = [
  {
    type: FormDefinitionType.SlovenskoSkGeneric,
    slovenskoSkUrl:
      'https://formulare.slovensko.sk/_layouts/eFLCM/DetailVzoruEFormulara.aspx?vid=00603481.stanoviskoKInvesticnemuZameru&vh=0&vl=8',
    slug: 'stanovisko-k-investicnemu-zameru',
    title: 'Žiadosť o stanovisko k investičnému zámeru',
    schemas: stanoviskoKInvesticnemuZameru,
    pospID: '00603481.stanoviskoKInvesticnemuZameru',
    pospVersion: '0.8',
    gestor: 'Martin Pinter',
    termsAndConditions: generalTermsAndConditions,
    messageSubjectDefault: 'Podanie',
    messageSubjectFormat:
      'e-ZST ž. {stavba.ulica} {stavba.nazov}, p.č. {stavba.parcelneCislo} kú {stavba.kataster}',
    ginisAssignment: {
      ginisOrganizationName: 'OUIC',
      ginisPersonName: 'Vícenová Marcela',
    },
    isSigned: false,
  },
  {
    type: FormDefinitionType.SlovenskoSkGeneric,
    slovenskoSkUrl:
      'https://formulare.slovensko.sk/_layouts/eFLCM/DetailVzoruEFormulara.aspx?vid=00603481.zavazneStanoviskoKInvesticnejCinnosti&vh=0&vl=7',
    slug: 'zavazne-stanovisko-k-investicnej-cinnosti',
    title: 'Žiadosť o záväzné stanovisko k investičnej činnosti',
    schemas: zavazneStanoviskoKInvesticnejCinnosti,
    pospID: '00603481.zavazneStanoviskoKInvesticnejCinnosti',

    pospVersion: '0.7',
    gestor: 'Martin Pinter',
    termsAndConditions: generalTermsAndConditions,
    messageSubjectDefault: 'Podanie',
    messageSubjectFormat:
      'e-SIZ ž. {stavba.ulica} {stavba.nazov}, p.č. {stavba.parcelneCislo} kú {stavba.kataster}',
    ginisAssignment: {
      ginisOrganizationName: 'OUIC',
      ginisPersonName: 'Vícenová Marcela',
    },
    isSigned: false,
  },
  {
    type: FormDefinitionType.SlovenskoSkGeneric,
    slovenskoSkUrl:
      'https://formulare.slovensko.sk/_layouts/eFLCM/DetailVzoruEFormulara.aspx?vid=00603481.predzahradky&vh=1&vl=0',
    slug: 'predzahradky',
    title: 'Predzáhradky',
    schemas: predzahradky,
    pospID: '00603481.predzahradky',
    pospVersion: '1.0',
    gestor: 'Martin Pinter',
    termsAndConditions: generalTermsAndConditions,
    messageSubjectFormat: 'Registrácia - {predzahradka.typRegistracie} predzáhradka',
    messageSubjectDefault: 'Registrácia predzáhradky',
    isSigned: false,
    ginisAssignment: {
      ginisOrganizationName: 'OUIC',
      ginisPersonName: 'Vícenová Marcela',
    },
  },
  {
    type: FormDefinitionType.SlovenskoSkGeneric,
    slovenskoSkUrl:
      'https://formulare.slovensko.sk/_layouts/eFLCM/DetailVzoruEFormulara.aspx?vid=00603481.komunitneZahrady&vh=1&vl=0',
    slug: 'komunitne-zahrady',
    title: 'Komunitné záhrady',
    schemas: komunitneZahrady,
    pospID: '00603481.komunitneZahrady',
    pospVersion: '1.0',
    gestor: 'Martin Pinter',
    termsAndConditions: generalTermsAndConditions,
    messageSubjectDefault: 'Žiadosť o komunitnú záhradu',
    messageSubjectFormat: 'Žiadosť o komunitnú záhradu - {pozemok.typPozemku} mestský pozemok',
    isSigned: false,
    ginisAssignment: {
      ginisOrganizationName: 'OUIC',
      ginisPersonName: 'Vícenová Marcela',
    },
  },
  {
    type: FormDefinitionType.SlovenskoSkTax,
    slovenskoSkUrl:
      'https://formulare.slovensko.sk/_layouts/eFLCM/DetailVzoruEFormulara.aspx?vid=esmao.eforms.bratislava.obec_024&vh=201501&vl=2',
    title: 'Priznanie k dani z nehnuteľností',
    slug: 'priznanie-k-dani-z-nehnutelnosti',
    schemas: priznanieKDaniZNehnutelnosti,
    pospID: 'esmao.eforms.bratislava.obec_024',
    pospVersion: '201501.2',
    gestor: 'Martin Pinter',
    termsAndConditions: taxTermsAndConditions,
    messageSubjectDefault: 'Priznanie k dani z nehnuteľností',
    isSigned: true,
  },
]
