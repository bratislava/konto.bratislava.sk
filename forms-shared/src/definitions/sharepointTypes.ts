export type SharepointColumnMapValue =
  | {
      type: 'json_path'
      info: string
    }
  | {
      type: 'mag_number' | 'title'
    }

export type SharepointRelationData = {
  databaseName: string
  originalTableId: string
  columnMap: Record<string, SharepointColumnMapValue>
}

export type SharepointData = {
  databaseName: string
  columnMap: Record<string, SharepointColumnMapValue>
  oneToMany?: Record<string, SharepointRelationData>
  oneToOne?: Record<string, SharepointRelationData>
}
