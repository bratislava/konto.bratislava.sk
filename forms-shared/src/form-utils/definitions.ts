import { FormDefinition, formDefinitions } from "../definitions/form-definitions";

export const getFormDefinitionBySlug = <T extends FormDefinition>(slug: string): T | null => {
  const formDefinition = formDefinitions.find(formDefinition => formDefinition.slug === slug)
  return formDefinition as T ?? null
}