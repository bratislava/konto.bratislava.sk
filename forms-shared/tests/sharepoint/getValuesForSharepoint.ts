import { SharepointColumnMapValue } from '../../src/definitions/sharepointTypes'
import {
  getArrayForOneToMany,
  getValueAtJsonPath,
  getValuesForFields,
} from '../../src/sharepoint/getValuesForSharepoint'

describe('getArrayForOneToMany', () => {
  it('should throw error if at the path there is no array', () => {
    try {
      getArrayForOneToMany(
        { jsonDataExtraDataOmitted: { val1: { val2: 'val3' } }, id: 'id' },
        'val1.val2',
      )
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }

    try {
      getArrayForOneToMany(
        { jsonDataExtraDataOmitted: { val1: { val2: { val3: 'val4' } } }, id: 'id' },
        'val1.val2',
      )
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }
  })

  it('should return the array at the path', () => {
    const array = [{ val1: '' }, { val2: '', val3: '' }, 'val3']
    const data = { val1: { val2: array } }
    const result = getArrayForOneToMany({ jsonDataExtraDataOmitted: data, id: 'id' }, 'val1.val2')
    expect(result).toEqual(array)
  })

  it('should return empty array for non-existing path', () => {
    const result = getArrayForOneToMany({ jsonDataExtraDataOmitted: {}, id: 'id' }, 'val1.val2')
    expect(result).toEqual([])
  })
})

describe('getValueAtJsonPath', () => {
  it('should throw error if at the path there is some object', () => {
    const data = { val1: { val2: 'val3' } }
    try {
      getValueAtJsonPath(data, 'val1')
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }
  })

  it('should throw error if there is array on non-primitives', () => {
    const dataArray = { val1: { val2: ['val3', 'val4', ['val5']] } }
    const data = { val1: { val2: ['val3', 'val4', { val5: 'val6' }] } }

    try {
      getValueAtJsonPath(data, 'val1.val2')
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }
    try {
      getValueAtJsonPath(dataArray, 'val1.val2')
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }
  })

  it('should return the value', () => {
    const data = { arr: ['val3', 'val4', 'val5'], bool: true, num: 1234, str: 'string' }
    expect(getValueAtJsonPath(data, 'arr')).toBe('val3, val4, val5')
    expect(getValueAtJsonPath(data, 'bool')).toBeTruthy()
    expect(getValueAtJsonPath(data, 'num')).toBe(1234)
    expect(getValueAtJsonPath(data, 'str')).toBe('string')
  })
})

describe('getValuesForFields', () => {
  it('should correctly fill in the data', () => {
    const data = { arr: ['val3', 'val4', 'val5'], bool: true, num: 1234, str: 'string' }
    const result = getValuesForFields(
      {
        databaseName: 'dbName',
        columnMap: {
          field1: {
            type: 'json_path',
            info: 'arr',
          },
          field2: {
            type: 'json_path',
            info: 'bool',
          },
          field3: {
            type: 'title',
          },
          field4: {
            type: 'mag_number',
          },
          field5: {
            type: 'json_path',
            info: 'non-existing.path',
          },
        },
      },
      {
        ginisDocumentId: 'ginisIdValue',
        formDefinitionSlug: 'formSlugValue',
        title: 'formTitle',
      },
      data,
      {
        field1: 'field1_col',
        field2: 'field2_col',
        field3: 'field3_col',
        field4: 'field4_col',
        field5: 'field5_col',
        field6: 'field6_col',
      },
      { field6: 'field6Val' },
    )

    expect(result).toEqual({
      field1_col: 'val3, val4, val5',
      field2_col: true,
      field3_col: 'formTitle',
      field4_col: 'ginisIdValue',
      field6_colId: 'field6Val',
    })
  })

  it('should throw error if there is unknown error', () => {
    const data = { arr: ['val3', 'val4', 'val5'], bool: true, num: 1234, str: 'string' }

    try {
      getValuesForFields(
        {
          databaseName: 'dbName',
          columnMap: {
            field1: {
              type: 'json_path',
              info: 'arr',
            },
            field2: {
              type: 'json_path',
              info: 'bool',
            },
            field3: {
              type: 'nonexistent',
            } as unknown as SharepointColumnMapValue,
            field4: {
              type: 'mag_number',
            },
            field5: {
              type: 'json_path',
              info: 'non-existing.path',
            },
          },
        },
        {
          ginisDocumentId: 'ginisIdValue',
          formDefinitionSlug: 'formSlugValue',
          title: 'formTitle',
        },
        data,
        {
          field1: 'field1_col',
          field2: 'field2_col',
          field3: 'field3_col',
          field4: 'field4_col',
          field5: 'field5_col',
          field6: 'field6_col',
        },
        { field6: 'field6Val' },
      )
      expect(true).toBeFalsy()
    } catch (error) {
      expect(error).toBeInstanceOf(TypeError)
    }
  })

  it('should throw error if there is nonexistent field', () => {
    const data = { arr: ['val3', 'val4', 'val5'], bool: true, num: 1234, str: 'string' }

    try {
      getValuesForFields(
        { databaseName: 'dbName', columnMap: {} },
        {
          ginisDocumentId: 'ginisIdValue',
          formDefinitionSlug: 'formSlugValue',
          title: 'formTitle',
        },
        data,
        {
          field1: 'field1_col',
          field2: 'field2_col',
          field3: 'field3_col',
          field4: 'field4_col',
          field5: 'field5_col',
          field6: 'field6_col',
        },
      )
      expect(true).toBeFalsy()
    } catch (error) {}
  })
})

// TODO snapshot tests
