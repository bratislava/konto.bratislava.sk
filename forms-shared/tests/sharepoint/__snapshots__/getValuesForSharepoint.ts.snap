// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getValuesForFields should match snapshots 1`] = `
{
  "DatumNarodenia": "1985-03-15",
  "Email": "jana.novakova@email.com",
  "GinisID": "MAG123",
  "MaximalnaVyskaNajomneho": 700,
  "MenoZiadatela": "Jana",
  "PreferovanaLokalita": "ruzinov, noveMesto",
  "PreferovanaVelkost": "3izbovy",
  "PriezviskoZiadatela": "Nováková",
  "TelefonneCislo": "+421901234567",
  "TrvalyPobytMesto": "Bratislava",
  "TrvalyPobytPsc": "81101",
  "TrvalyPobytUlicaACislo": "Hlavná 123",
}
`;

exports[`getValuesForFields should match snapshots 2`] = `
{
  "ByvanieVMestskomNajomnomByte": false,
  "ChronickeOchorenie": true,
  "DovodyPodaniaZiadosti": "Potrebujeme väčší byt pre našu rastúcu rodinu a lepšie podmienky pre dieťa so zdravotným postihnutím.",
  "GinisID": "MAG123",
  "KontaktovanyEmailom": true,
  "MaximalnaVyskaNajomneho": 700,
  "PreferovanaLokalita": "ruzinov, noveMesto",
  "PreferovanaVelkost": "3izbovy",
  "RizikoveFaktory": true,
  "Vek": "menejAko63",
  "ZiadatelBezbarierovyByt": false,
  "ZiadatelBytovaNudza": true,
  "ZiadatelDatumNarodenia": "1985-03-15",
  "ZiadatelDlzkaBytovejNudze": "menejAko1",
  "ZiadatelEmail": "jana.novakova@email.com",
  "ZiadatelExistujuceDiagnozy": "astma, chronickaParadentoza",
  "ZiadatelFunkcnaPorucha": false,
  "ZiadatelInePrijmyVyska": 150,
  "ZiadatelMeno": "Jana",
  "ZiadatelPriezvisko": "Nováková",
  "ZiadatelRodinnyStav": "rozvedeny",
  "ZiadatelRodnePriezvisko": "Kováčová",
  "ZiadatelSkutocnyPobytRovnakyAkoT": true,
  "ZiadatelStatnaPrislusnost": "slovenska",
  "ZiadatelTelefonneCislo": "+421901234567",
  "ZiadatelTrvalyPobytMesto": "Bratislava",
  "ZiadatelTrvalyPobytPobytMenejAko": true,
  "ZiadatelTrvalyPobytPsc": "81101",
  "ZiadatelTrvalyPobytUlicaACislo": "Hlavná 123",
  "ZiadatelTrvalyPobytVlastnikNehnu": false,
  "ZiadatelTypSkutocnehoByvania": "institucionalnaStarostlivost",
  "ZiadatelVyzivneVyska": 200,
  "ZiadatelZamestnaniePrijem": 1200,
  "ZoznamRizikovychFaktorov": "opustenieUstavnejStarostlivosti, osamelyRodic",
}
`;

exports[`getValuesForFields should match snapshots 3`] = `
{
  "GinisID": "MAG123",
  "SucasneByvaniaRovnakaSituacia": true,
  "TzpPreukaz": true,
  "VlastnikNehnutelnosti": false,
  "ZiadatelDatumNarodenia": "1982-07-22",
  "ZiadatelMeno": "Peter",
  "ZiadatelMieraFunkcnejPoruchy": "50az74",
  "ZiadatelPriezvisko": "Horváth",
  "ZiadatelRodinnyStav": "slobodny",
  "ZiadatelRodnePriezvisko": "Horváth",
  "ZiadatelSamostatnaZarobkovaCinno": 300,
  "ZiadatelSkutocnyPobytMesto": "Bratislava",
  "ZiadatelSkutocnyPobytPsc": "82105",
  "ZiadatelSkutocnyPobytUlica": "Nová 456",
  "ZiadatelStatnaPrislusnost": "slovenska",
  "ZiadatelZamestnaniePrijem": 1500,
}
`;

exports[`getValuesForFields should match snapshots 4`] = `
{
  "ChronickeOchorenie": false,
  "GinisID": "MAG123",
  "MaPrijem": false,
  "Student": true,
  "SucasneByvanieRovnakeAkoVase": true,
  "TzpPreukaz": false,
  "VlastnikNehnutelnosti": false,
  "ZiadatelDatumNarodenia": "2010-11-30",
  "ZiadatelMeno": "Lucia",
  "ZiadatelPriezvisko": "Nováková",
  "ZiadatelStatnaPrislusnost": "slovenska",
}
`;

exports[`getValuesForFields should match snapshots 5`] = `
{
  "GinisID": "MAG123",
  "MaPrijem": true,
  "MieraFunkcnejPoruchy": "75az100",
  "Student": true,
  "SucasneByvanieRovnakeAkoVase": true,
  "TzpPreukaz": true,
  "VlastnikNehnutelnosti": false,
  "ZiadatelDatumNarodenia": "2005-04-18",
  "ZiadatelMeno": "Michal",
  "ZiadatelPriezvisko": "Novák",
  "ZiadatelStatnaPrislusnost": "slovenska",
  "ZiadatelZamestnaniePrijem": 200,
}
`;

exports[`getValuesForFields should match snapshots 6`] = `
{
  "DochodokVyska": 550,
  "GinisID": "MAG123",
  "MieraFunkcnejPoruchy": "50az74",
  "SucasneByvanieRovnakeAkoVase": true,
  "TzpPreukaz": true,
  "VlastnikNehnutelnosti": false,
  "ZiadatelDatumNarodenia": "1960-09-05",
  "ZiadatelMeno": "Mária",
  "ZiadatelPriezvisko": "Kováčová",
  "ZiadatelStatnaPrislusnost": "slovenska",
}
`;
